test/end2end_test.go

```diff
@@ -421,6 +448,7 @@ func (e env) dialer(addr string, timeout time.Duration) (net.Conn, error) {
 var (
 	tcpClearEnv   = env{name: "tcp-clear-v1-balancer", network: "tcp", balancer: "v1"}
 	tcpTLSEnv     = env{name: "tcp-tls-v1-balancer", network: "tcp", security: "tls", balancer: "v1"}
+	tcpLocalEnv   = env{name: "tcp-local-v1-balancer", network: "tcp", listenerAddr: "[::1]:0", security: "local", balancer: "v1"}
```

Comment:
There is now no test that uses this.  (And I think that's OK.)

Please add an end-to-end test specifically for the local credentials.  It should *ideally*:

1. use local creds on client and server with a TCP dial to "localhost" that succeeds
1. use local creds on client and server with a UDS connection that succeeds
1. use local creds on client and server with a loopback-but-non-127.* TCP dial that fails (observe failure on client, then use WithInsecure on client and observe failure on server)

I'm not sure how to do that last one in a portable way, but it should be possible if you limit it to linux systems.

I think you can just do all this ~manually using a `stubServer`, and not modify the global `testServer`.